core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionVerificationServiceSpec.js

```diff
@@ -17,183 +17,204 @@
-              }]
-            }],
-            default_outcome: {
-              dest: 'default',
+  beforeEach(inject(function($rootScope, $controller, $injector) {
+    scope = $rootScope.$new();
+    ecs = $injector.get('editorContextService');
+    cls = $injector.get('changeListService');
+    ess = $injector.get('explorationStatesService');
+    siis = $injector.get('stateInteractionIdService');
+    scas = $injector.get('stateCustomizationArgsService');
+    idc = $injector.get('interactionDetailsCache');
+    sof = $injector.get('SolutionObjectFactory');
+    svs = $injector.get('SolutionVerificationService');
+    IS = $injector.get('INTERACTION_SPECS');
+    $httpBackend = $injector.get('$httpBackend');
+    scope.stateInteractionIdService = siis;
+    scope.stateCustomizationArgsService = scas;
+    scope.interactionDetailsCache = idc;
+
+    ess.init({
+      'First State': {
+        content: {
+          html: 'First State Content',
+          audio_translations: {}
+        },
+        interaction: {
+          id: 'TextInput',
+          answer_groups: [{
+            correct: false,
+            outcome: {
+              dest: 'End State',
               feedback: [],
               param_changes: []
             },
-            fallbacks: [],
-            hints: [{
-              hint_text: 'one'
-            }, {
-              hint_text: 'two'
+            rule_specs: [{
+              inputs: {x: 'abc'},
+              rule_type: 'Contains'
             }]
+          }],
+          default_outcome: {
+            dest: 'First State',
+            feedback: [],
+            param_changes: []
           },
-          param_changes: []
+          fallbacks: [],
+          hints: [{
+            hint_text: 'one'
+          }, {
+            hint_text: 'two'
+          }]
         },
-        'End State': {
-          content: {
-            html: '',
-            audio_translations: {}
-          },
-          interaction: {
-            id: 'TextInput',
-            answer_groups: [{
-              rule_specs: [],
-              outcome: {},
-              correct: false
-            }],
-            default_outcome: {
-              dest: 'default',
-              feedback: [],
-              param_changes: []
-            },
-            fallbacks: [],
-            hints: []
+        param_changes: []
+      },
+      'End State': {
+        content: {
+          html: '',
+          audio_translations: {}
+        },
+        interaction: {
+          id: 'TextInput',
+          answer_groups: [{
+            rule_specs: [],
+            outcome: {},
+            correct: false
+          }],
+          default_outcome: {
+            dest: 'default',
+            feedback: [],
+            param_changes: []
           },
-          param_changes: []
+          fallbacks: [],
+          hints: []
+        },
+        param_changes: []
+      }
+    });
+
+    var stateEditorCtrl = $controller('StateEditor', {
```

Comment:
This is supposed to be testing a service. I'm wondering why you need all these controllers. I think something is wrong if you need to rely on controllers when testing a service...

Also, if you need to mock functions in dependent services in order to stop everything depending on everything, go ahead.